### PR TITLE
libaec: init at 1.0.4; zopfli: include all headers

### PR DESCRIPTION
**Pull request by tbenst**
###### Motivation for this change
Add/fix imagecodecs dependencies

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bobvanderlinden** [comment]
I have very limited connectivity at the moment, so I won't be able to test this in the next few weeks.

**Comment by adisbladis** [comment]
@GrahamcOfBorg build libaec zopfli

File: pkgs/tools/compression/zopfli/default.nix [hunk 1]
```diff
@@ -19,6 +19,7 @@ stdenv.mkDerivation rec {
 
   postInstall = ''
     install -Dm444 -t $out/share/doc/zopfli ../README*
+    cp $src/src/zopfli/*.h $dev/include/
```
Comment (adisbladis) [inline pkgs/tools/compression/zopfli/default.nix:22]: This doesn't actually seem to make any difference?
Note: I change this from `$out` to `$dev` as headers should not be in the runtime closure.

File: pkgs/development/libraries/libaec/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libaec";
+  version  = "1.0.4";
+
+  src = fetchzip {
+    url = "https://gitlab.dkrz.de/k202009/libaec/-/archive/v${version}/${pname}-v${version}.tar.gz";
+    sha256 = "1rpma89i35ahbalaqz82y201syxni7jkf9892jlyyrhhrvlnm4l2";
+  };
```
Comment (jtojnar) [inline pkgs/development/libraries/libaec/default.nix:12]:
```suggestion
  src = fetchFromGitLab {
    domain = "gitlab.dkrz.de";
    owner = "k202009";
    repo = "libaec";
    rev = "v${version}";
    sha256 = "1rpma89i35ahbalaqz82y201syxni7jkf9892jlyyrhhrvlnm4l2";
  };
```

File: pkgs/tools/compression/zopfli/default.nix (hunk 1 above)
Comment (tbenst) [reply pkgs/tools/compression/zopfli/default.nix:22]: @adisbladis right now, the derivation does not include any of these headers: https://github.com/google/zopfli/tree/master/src/zopfli

File: pkgs/development/libraries/libaec/default.nix (hunk 2 above)
Comment (tbenst) [reply pkgs/development/libraries/libaec/default.nix]: done, thx!

File: pkgs/tools/compression/zopfli/default.nix (hunk 1 above)
Comment (tbenst) [reply pkgs/tools/compression/zopfli/default.nix:22]: (some of these headers are needed for imagecodecs)

Comment (adisbladis) [reply pkgs/tools/compression/zopfli/default.nix:22]: My bad, I misread things.

**Pull request merged**